src/toast/ops/conviqt.py

```diff
@@ -658,3 +658,97 @@ def _exec(self, data, detectors=None, **kwargs):
                 timer.report_clear(f"conviqt process detector {det}")
 
         return
+
+
+
+class SimTEBConviqt(SimConviqt):
+    """
+    Operator that uses libconviqt to generate beam-convolved timestreams.
+    This operator should be used in presence of a spinning HWP which makes the beam time-dependent,
+    constantly mapping the co- and cross-polar responses on to each other.
+    In the parent class OpSimConviqt we assume the beam to be static.
+
+
+    The convolution  is performed by  coupling each IQU component of the signal propertly as:
+    :math:`skyT_lm * beamT_lm, skyE_lm * Re{P}, skyB_lm * Im{P}`.
+
+    For extra details please refer to [this note ](https://giuspugl.github.io/reports/Notes_TEB_convolution.html)
+
+"""
+
+    @function_timer
+    def _exec(self, data, detectors=None, **kwargs):
+        if not self.available:
+            raise RuntimeError("libconviqt is not available")
+
+        if self.comm is None:
+            raise RuntimeError("libconviqt requires MPI")
+
+        if self.detector_pointing is None:
+            raise RuntimeError("detector_pointing cannot be None.")
+
+        log = Logger.get()
+
+        timer = Timer()
+        timer.start()
+
+        # Expand detector pointing
+        self.detector_pointing.apply(data, detectors=detectors)
+
+        all_detectors = self._get_all_detectors(data, detectors)
+
+        for det in all_detectors:
+            verbose = self.comm.rank == 0 and self.verbosity > 0
+
+            # Expand detector pointing
+            self.detector_pointing.apply(data, detectors=[det])
+
+            if det in self.sky_file_dict:
+                sky_file = self.sky_file_dict[det]
+            else:
+                sky_file = self.sky_file.format(detector=det, mc=self.mc)
+            sky = self.get_sky(sky_file, det, verbose)
+
+            beam_file = self.beam_file.format(detector=det, mc=self.mc)
+            beam_T = self.get_beam(beam_file.replace(".fits","_T.fits"), det, verbose)
+            beam_E = self.get_beam(beam_file.replace(".fits","_E.fits"), det, verbose)
+            beam_B = self.get_beam(beam_file.replace(".fits","_B.fits"), det, verbose)
+
+            detector = self.get_detector(det)
+
+            theta, phi, psi, psi_pol = self.get_pointing(data, det, verbose)
+
+            # I-beam convolution
+            pnt = self.get_buffer(theta, phi, psi, det, verbose)
+
+            convolved_data = self.convolve(sky, beam_T, detector, pnt, det, verbose)
+
+            del pnt,
+            # Q-beam convolution
+            pnt = self.get_buffer(theta, phi, psi, det, verbose)
+            convolved_data += np.cos(2 * psi_pol) * self.convolve(
```

Comment:
I am confused about this line here. The note linked in the description says that
``` Q + iU = Sum[(b^E + ib^B)(a^E + ia^B)] * D ```
which implies that
```
Q = Sum[b^E * a^E - b^B * a^B]
U = Sum[b^E * a^B + b^B * a^E]
```
but on this line we seem to be evaluating `Sum[b^E * (a^E + ia^B)]` -- a complex-valued quantity. The issue may be completely semantical but I propose you implement a unit test that compares the standard conviqt operator to this new one. In the absence of a HWP, they should agree.

Reply:
This is a good point! IT might also linked to https://github.com/hpc4cmb/toast/pull/534#pullrequestreview-869602524 
The way, I implemented this convolution indeed is  as you suggest, meaning that the beams i use in the convolution of Q and U signals are respectively 
```
beam_E = [0, b^E , -b^B ]
beam_B = [0, b^B,b^E]
```

Reply:
see https://github.com/hpc4cmb/toast/blob/73c758831f3f694ef1d483e53f02de2f928bb929/src/toast/tests/ops_sim_tod_conviqt.py#L113
and https://github.com/hpc4cmb/toast/blob/73c758831f3f694ef1d483e53f02de2f928bb929/src/toast/tests/ops_sim_tod_conviqt.py#L126